lib/matplotlib/axes/_subplots.py

```diff
@@ -172,8 +172,8 @@ def _make_twin_axes(self, *kl, **kwargs):
         Make a twinx axes of self. This is used for twinx and twiny.
         """
         from matplotlib.projections import process_projection_requirements
-        if 'sharex' in kwargs and 'sharey' in kwargs:
-            raise ValueError("Twinned Axes may share only one axis.")
+        if kwargs["sharex"] is not self and kwargs["sharey"] is not self:
```

Comment:
This isn't the same as the old check at all 😉

Reply:
hum, result is identical unless one of the 2 sharex/sharey is "self", isn't it ?